Orange/data/table/base.py

```diff
@@ -0,0 +1,1409 @@
+            return [v if v in result.columns else v.copy(new_name=v.name + suffix) for v in varlist]
+
+        # dedup because the target join valriable doesn't get renamed and there is
+        # only one column, wehile without this, the domain would have two
+        def dedup_inorder(varlist):
+            s = set()
+            return [x for x in varlist if not (x in s or s.add(x))]
+
+        # merge domain
+        new_domain = Domain(
+            dedup_inorder(suffix_dups(self.domain.attributes, suffix_left) +
+                          suffix_dups(right.domain.attributes, suffix_right)),
+            dedup_inorder(suffix_dups(self.domain.class_vars, suffix_left) +
+                          suffix_dups(right.domain.class_vars, suffix_right)),
+            dedup_inorder(suffix_dups(self.domain.metas, suffix_left) +
+                          suffix_dups(right.domain.metas, suffix_right))
+        )
+        result.domain = new_domain
+
+        # always reindex
+        result.index = self._new_id(len(result), force_list=True)
+
+        return result
+
+    def approx_len(self):
+        """Return the approximate length of the table."""
+        # TODO: remove, here solely for SQL compatibility, which would ideally be replaced with Spark
+        # https://github.com/biolab/orange3/wiki/pandas-migration#future-todos-and-architectural-wishlist
+        return len(self)
+
+    def exact_len(self):
+        """Return the exact length of the table."""
+        # TODO: remove, here solely for SQL compatibility, which would ideally be replaced with Spark
+        # https://github.com/biolab/orange3/wiki/pandas-migration#future-todos-and-architectural-wishlist
+        return len(self)
+
+    def has_missing(self):
+        """Return `True` if there are any missing attribute or class values."""
+        # manual access to columns because dumping to a numpy array (with self.X) is slower
+        return self[self.domain.attributes].isnull().any().any() or self.has_missing_class()
+
+    def has_missing_class(self):
+        """Return `True` if there are any missing class values."""
+        return self[self.domain.class_vars].isnull().any().any()
+
+    def iterrows(self):
+        """An override to return TableSeries instead of Series (pandas doesn't do that by default)."""
+        # super here is the next item in the MRO, e.g. pd.DataFrame or pd.SparseDataFrame
+        gen = super().iterrows()
+        for item in gen:
+            yield (item[0], self._constructor_sliced(item[1]))
+
+    @deprecated
+    def checksum(self, include_metas=True):
+        """Return a checksum over X, Y, metas and W."""
+        # TODO: zlib.adler32 does not work for numpy arrays with dtype object
+        # (after pickling and unpickling such arrays, checksum changes)
+        # Why, and should we fix it or remove it?
+        cs = zlib.adler32(np.ascontiguousarray(self.X))
+        cs = zlib.adler32(np.ascontiguousarray(self.Y), cs)
```

Comment:
This method should probably be renamed to:
```py
def __hash__(self):
    return hash(bytes(self.values))
```
Will work for non-numeric objects in `metas` just equally invalid.